client/client_test.go

```diff
@@ -328,7 +328,7 @@ func TestHTTPClusterClientDo(t *testing.T) {
 		// context.DeadlineExceeded short-circuits Do
 		{
 			client: &httpClusterClient{
-				endpoints: []url.URL{fakeURL, fakeURL},
+				endpoints: []url.URL{fakeURL},
```

Comment:
The previous test is not correct after this commit. If it has two endpoints, we should expect it try the second one before returning error. Use single endpoint to fix the test.